nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/DebugTask.java

```diff
@@ -128,85 +147,114 @@ private List<Configuration> createNewSeedsToRetry() {
+            long end = range.getRight();
+
+            if (start < end) {
+                Logger.getGlobal().log(Level.INFO, "Debugging binary for " + this.test + " " + start + " : " + end);
+
+                boolean binarySuccess = false;
+                long midPoint = (start + end) / 2;
+                if ((failingConfiguration = this.failsWithConfig(config, start, midPoint)) != null) {
+                    pairs.add(Pair.of((Pair<Long, Long>)Pair.of(start, midPoint), failingConfiguration));
+                    binarySuccess = true;
+                }
+                if ((failingConfiguration = this.failsWithConfig(config, midPoint + 1, end)) != null) {
+                    pairs.add(Pair.of((Pair<Long, Long>)Pair.of(midPoint + 1, end), failingConfiguration));
+                    binarySuccess = true;
+                }
 
-            long midPoint = (start + end) / 2;
-            if ((failingConfiguration = this.failsWithConfig(config, start, midPoint)) != null) {
-                end = midPoint;
-                continue;
-            } else if ((failingConfiguration = this.failsWithConfig(config, midPoint + 1, end)) != null) {
-                start = midPoint + 1;
-                continue;
+                // If both halves fail, try the entire range
+                if (!binarySuccess) {
+                    Logger.getGlobal().log(Level.SEVERE, "Binary splitting did not work. Going to linear");
+                    allFailingConfigurations.addAll(this.startDebugLinear(config, start, end));
+                }
             } else {
-                Logger.getGlobal().log(Level.FINE, "Binary splitting did not work. Going to linear");
-                failingConfiguration = this.startDebugLinear(config, start, end);
-                break;
+                // Since start <= end is always true, this branch means start == end, so reached end
+                if (failingConfiguration != null) {
+                    allFailingConfigurations.add(this.reportDebugInfo(failingConfiguration));
+                }
             }
         }
-        if (failingConfiguration != null) {
-            return this.reportDebugInfo(failingConfiguration);
-        }
-        return failingConfiguration;
+
+        return allFailingConfigurations;
     }
 
     private Configuration reportDebugInfo(Configuration failingConfiguration) {
         return this.failsWithConfig(failingConfiguration, failingConfiguration.start, failingConfiguration.end, true);
     }
 
-    public Configuration startDebugLinear(Configuration config, long start, long end) {
+    public List<Configuration> startDebugLinear(Configuration config, long start, long end) {
+        List<Configuration> allFailingConfigurations = new LinkedList<Configuration>();
+
+        List<Pair<Pair<Long, Long>, Configuration>> pairs = new LinkedList<Pair<Pair<Long, Long>, Configuration>>();
+        pairs.add((Pair<Pair<Long, Long>, Configuration>)Pair.of((Pair<Long, Long>)Pair.of(start, end),
```

Comment:
Same thing here about initial configuration in the list.

Reply:
Thanks for pointing that out!